erc20-price/src/lib.rs

```diff
@@ -39,3 +48,91 @@ fn map_price_for_tokens(
 
     Ok(prices)
 }
+
+#[substreams::handlers::store]
+fn store_chainlink_aggregator(block: eth::Block, output: StoreSetProto<Aggregator>) {
+    for call in block.calls() {
+        if let Some(decoded_call) = price_feed::functions::ConfirmAggregator::match_and_decode(call)
+        {
+            let decimals = chainlink_aggregator::functions::Decimals {}
+                .call(decoded_call.aggregator.to_vec())
+                .unwrap_or(BigInt::zero());
+            let description = chainlink_aggregator::functions::Description {}
+                .call(decoded_call.aggregator.to_vec())
+                .unwrap_or(String::from(""));
+
+            let base_quote: Vec<&str> = description.split(" / ").collect();
+
+            if base_quote.len() != 2 {
+                log::info!(
+                    "[ChainlinkAggregator] Unexpected Description: {}",
+                    description
+                );
+                continue;
+            }
+
+            let mut aggregator_address = decoded_call.aggregator;
+
+            let nested_aggregator = chainlink_aggregator::functions::Aggregator {}
+                .call(aggregator_address.to_vec())
+                .unwrap_or(Vec::<u8>::new());
+
+            if nested_aggregator.is_empty().not() {
+                // In `AggregatorFacade` contracts, the aggregator contract is nested two times.
+                aggregator_address = nested_aggregator;
+            }
+
+            let aggregator = Aggregator {
+                address: address_pretty(&aggregator_address).to_string(),
```

Comment:
it's unnecessary to pretty the address during storage and data passing

Reply:
Noted!